src/DynamicData.Tests/Cache/TransformAsyncFixture.cs

```diff
@@ -207,6 +205,84 @@ public void Update()
         stub.Results.Messages[1].Updates.Should().Be(1, "Should be 1 update");
     }
 
+
+    private record PersonWithAgeGroup(Person Person, string AgeGroup);
+
+
+
+    [Fact]
+    public void TransformWithoutRefresh()
+    {
+        using var source = new SourceCache<Person, string>(p => p.Name);
+        using var results = source.Connect()
+            .AutoRefresh()
+            .TransformAsync((p, key) => Task.FromResult(new PersonWithAgeGroup(p, p.Age < 18 ? "Child" : "Adult"))).AsAggregator();
+
+        var person = new Person("SomeOne", 16);
+        source.AddOrUpdate(person);
+
+        results.Data.Count.Should().Be(1);
+        results.Data.Lookup("SomeOne").Value.AgeGroup.Should().Be("Child");
+
+        person.Age = 21;
+
+
+        results.Data.Count.Should().Be(1);
+        results.Data.Lookup("SomeOne").Value.AgeGroup.Should().Be("Child");
+
+    }
+
+    [Fact]
+    public void TransformOnRefresh()
+    {
+        using var source = new SourceCache<Person, string>(p => p.Name);
+        using var results = source.Connect()
+            .AutoRefresh()
+            .TransformAsync((p, key) => Task.FromResult(new PersonWithAgeGroup(p, p.Age < 18  ? "Child" : "Adult")), TransformAsyncOptions.WithTransformOnRefresh).AsAggregator();
+
+        var person = new Person("SomeOne", 16);
+        source.AddOrUpdate(person);
+
+        results.Data.Count.Should().Be(1);
+        results.Data.Lookup("SomeOne").Value.AgeGroup.Should().Be("Child");
+        
+        person.Age = 21;
+
+
+        results.Data.Count.Should().Be(1);
+        results.Data.Lookup("SomeOne").Value.AgeGroup.Should().Be("Adult");
+
+    }
+
+    
+    [Theory, InlineData(10), InlineData(100)]
+
+    public async Task WithMaxConcurrency(int maxConcurrency)
+    {
+        /* We need to test whether the max concurrency has any effect.
```

Comment:
@JakenVeina any ideas how I can test concurrency apart from looking at the timings?